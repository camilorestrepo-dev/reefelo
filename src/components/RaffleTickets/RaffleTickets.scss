@import "../../styles/shared.scss";

.raffle-tickets-container {
  display: grid;
  grid-template-columns: repeat(10, 1fr);
  gap: $spacing-8;

  .raffle-ticket-card {
    border: 1px solid $grey;
    border-radius: $radius-5;
    aspect-ratio: 1;
    font-size: x-small;
    display: flex;
    justify-content: center;
    align-items: center;
    background: $enabled-ticket;
    padding: 0;
    cursor: pointer;

    &.disabled-ticket {
      position: relative;
      background: $disabled-ticket;

      &.payed-ticket {
        &::before {
          content: '';
          width: 2px;
          height: 100%;
          background: $text-color;
          position: absolute;
          left: 50%;
          top: 50%;
          transform: translate(-50%, -50%) rotate(-45deg);
        }

        &::after {
          content: '';
          width: 2px;
          height: 100%;
          background: $text-color;
          position: absolute;
          left: 50%;
          top: 50%;
          transform: translate(-50%, -50%) rotate(45deg);
        }
      }
    }
  }
}

@media (max-width: 768px) {
  .raffle-tickets-container {
    &.small-view {
      grid-template-columns: repeat(5, 1fr);
    }
  }
}

@media (min-width: 768px) {
  .raffle-tickets-container {
    .raffle-ticket-card {
      font-size: medium;
    }
  }
}

@media (min-width: 1024px) {
  .raffle-tickets-container {
    .raffle-ticket-card {
      font-size: large;
    }
  }
}