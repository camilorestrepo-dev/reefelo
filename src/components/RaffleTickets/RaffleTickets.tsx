import "./RaffleTickets.scss";
import classnames from "classnames";
import { TicketData } from "../../models/ticket";

type OnTicketPressed = (id: string) => void;

type RaffleTicketsProps = {
  tickets: Array<TicketData>;
  onTicketPressed: OnTicketPressed;
};

const RaffleTickets: React.FC<RaffleTicketsProps> = ({
  tickets,
  onTicketPressed,
}: RaffleTicketsProps) => {
  return (
    <div className={classnames("raffle-tickets-container", { "small-view": tickets.length == 10 })}>
      {tickets.map((ticket) => {
        const { id, enabled, payed } = ticket;
        return (
          <label
            className={classnames("raffle-ticket-card", {
              "disabled-ticket": !enabled,
              "payed-ticket": !enabled && payed,
            })}
            key={id}
            data-testid={`ticket-btn-${id}`}
            onClick={() => onTicketPressed(id)}
          >
            {id}
          </label>
        );
      })}
    </div>
  );
};

export default RaffleTickets;
