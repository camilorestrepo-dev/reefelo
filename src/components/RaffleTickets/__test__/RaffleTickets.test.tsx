import { render, screen, fireEvent } from "@testing-library/react";
import RaffleTickets from "../RaffleTickets";
import props from "./porps.json";

test("render correctly", () => {
  const mockFn = () => {};
  const component = render(<RaffleTickets tickets={props} onTicketPressed={mockFn} />);
  expect(component.container).toMatchSnapshot();
});

test("click in the right ticket", async () => {
  let pressedTicket: string;
  const onPressed = (id: string) => {
    pressedTicket = id;
  };
  render(<RaffleTickets tickets={props} onTicketPressed={onPressed} />);

  const firstTicket = screen.getByTestId("ticket-btn-00");
  expect(firstTicket).toBeInTheDocument();

  fireEvent.click(firstTicket);
  expect(pressedTicket).toBe("00");
});
