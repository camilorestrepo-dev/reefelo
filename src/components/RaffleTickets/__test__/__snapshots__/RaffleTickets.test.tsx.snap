// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div>
  <div
    class="raffle-tickets-container"
  >
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-00"
    >
      00
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-01"
    >
      01
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-02"
    >
      02
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-03"
    >
      03
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-04"
    >
      04
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-05"
    >
      05
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-06"
    >
      06
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-07"
    >
      07
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-08"
    >
      08
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-09"
    >
      09
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-10"
    >
      10
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-11"
    >
      11
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-12"
    >
      12
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-13"
    >
      13
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-14"
    >
      14
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-15"
    >
      15
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-16"
    >
      16
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-17"
    >
      17
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-18"
    >
      18
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-19"
    >
      19
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-20"
    >
      20
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-21"
    >
      21
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-22"
    >
      22
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-23"
    >
      23
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-24"
    >
      24
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-25"
    >
      25
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-26"
    >
      26
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-27"
    >
      27
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-28"
    >
      28
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-29"
    >
      29
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-30"
    >
      30
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-31"
    >
      31
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-32"
    >
      32
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-33"
    >
      33
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-34"
    >
      34
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-35"
    >
      35
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-36"
    >
      36
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-37"
    >
      37
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-38"
    >
      38
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-39"
    >
      39
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-40"
    >
      40
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-41"
    >
      41
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-42"
    >
      42
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-43"
    >
      43
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-44"
    >
      44
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-45"
    >
      45
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-46"
    >
      46
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-47"
    >
      47
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-48"
    >
      48
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-49"
    >
      49
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-50"
    >
      50
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-51"
    >
      51
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-52"
    >
      52
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-53"
    >
      53
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-54"
    >
      54
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-55"
    >
      55
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-56"
    >
      56
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-57"
    >
      57
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-58"
    >
      58
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-59"
    >
      59
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-60"
    >
      60
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-61"
    >
      61
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-62"
    >
      62
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-63"
    >
      63
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-64"
    >
      64
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-65"
    >
      65
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-66"
    >
      66
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-67"
    >
      67
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-68"
    >
      68
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-69"
    >
      69
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-70"
    >
      70
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-71"
    >
      71
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-72"
    >
      72
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-73"
    >
      73
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-74"
    >
      74
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-75"
    >
      75
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-76"
    >
      76
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-77"
    >
      77
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-78"
    >
      78
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-79"
    >
      79
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-80"
    >
      80
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-81"
    >
      81
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-82"
    >
      82
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-83"
    >
      83
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-84"
    >
      84
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-85"
    >
      85
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-86"
    >
      86
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-87"
    >
      87
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-88"
    >
      88
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-89"
    >
      89
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-90"
    >
      90
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-91"
    >
      91
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-92"
    >
      92
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-93"
    >
      93
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-94"
    >
      94
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-95"
    >
      95
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-96"
    >
      96
    </label>
    <label
      class="raffle-ticket-card"
      data-testid="ticket-btn-97"
    >
      97
    </label>
    <label
      class="raffle-ticket-card disabled-ticket"
      data-testid="ticket-btn-98"
    >
      98
    </label>
    <label
      class="raffle-ticket-card disabled-ticket payed-ticket"
      data-testid="ticket-btn-99"
    >
      99
    </label>
  </div>
</div>
`;
